service/src/main/kotlin/fi/espoo/evaka/messaging/bulletin/BulletinQueries.kt

```diff
@@ -126,28 +124,53 @@ fun Database.Transaction.sendBulletin(
     // language=sql
     val insertBulletinInstancesSql = """
         WITH children AS (
-            SELECT pl.child_id, br.id AS bulletin_receiver_id
+            SELECT child_id, bulletin_id FROM bulletin_receiver
+            WHERE
+            bulletin_id = :bulletinId 
+            AND unit_id IS NOT NULL 
+            AND group_id IS NOT NULL 
+            AND child_id IS NOT NULL
+            
+            UNION DISTINCT
+            
+            SELECT pl.child_id, br.bulletin_id
             FROM bulletin b
             JOIN bulletin_receiver br ON br.bulletin_id = b.id 
             JOIN daycare_group dg ON br.group_id = dg.id
             JOIN daycare_group_placement gpl ON dg.id = gpl.daycare_group_id AND daterange(gpl.start_date, gpl.end_date, '[]') @> :date
             JOIN placement pl ON gpl.daycare_placement_id = pl.id
-            WHERE b.id = :bulletinId
+            WHERE b.id = :bulletinId 
+            AND br.unit_id IS NOT NULL 
+            AND br.group_id IS NOT NULL 
+            AND br.child_id IS NULL
+            
+            UNION DISTINCT
+            
+            SELECT pl.child_id, br.bulletin_id
+            FROM bulletin b
+            JOIN bulletin_receiver br ON br.bulletin_id = b.id
+            JOIN daycare_group dg ON br.unit_id = dg.daycare_id
+            JOIN daycare_group_placement gpl ON dg.id = gpl.daycare_group_id AND daterange(gpl.start_date, gpl.end_date, '[]') @> :date
+            JOIN placement pl ON gpl.daycare_placement_id = pl.id
```

Comment:
we could forget groups and group placements here and just check that the child has placement in the unit.

FROM bulletin b
JOIN bulletin_receiver br ON br.bulletin_id = b.id
JOIN placement pl ON br.unit_id = pl.unit_id AND daterange(pl.start_date, pl.end_date, '[]') @> :date